tests/www/views/test_views_dataset.py

```diff
@@ -0,0 +1,281 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import string
+
+import pendulum
+import pytest
+from dateutil.tz import UTC
+
+from airflow import Dataset
+from airflow.models.dataset import DatasetEvent, DatasetModel
+from airflow.operators.empty import EmptyOperator
+from airflow.utils import timezone
+from tests.test_utils.asserts import assert_queries_count
+from tests.test_utils.db import clear_db_datasets
+
+
+class TestDatasetEndpoint:
+
+    default_time = "2020-06-11T18:00:00+00:00"
+
+    @pytest.fixture(autouse=True)
+    def cleanup(self):
+        clear_db_datasets()
+        yield
+        # clear_db_datasets()
+
+
+class TestGetDatasets(TestDatasetEndpoint):
+    def test_should_respond_200(self, admin_client, session):
+        datasets = [
+            DatasetModel(
+                id=i,
+                uri=f"s3://bucket/key/{i}",
+                extra={"foo": "bar"},
+                created_at=timezone.parse(self.default_time),
+                updated_at=timezone.parse(self.default_time),
+            )
+            for i in [1, 2]
+        ]
+        session.add_all(datasets)
+        session.commit()
+        assert session.query(DatasetModel).count() == 2
+
+        with assert_queries_count(8):
+            response = admin_client.get("/object/list_datasets")
+
+        assert response.status_code == 200
+        response_data = response.json
+        assert response_data == {
+            "datasets": [
+                {
+                    "id": 1,
+                    "uri": "s3://bucket/key/1",
+                    "last_dataset_update": None,
+                    "total_updates": 0,
+                    "producing_task_count": 0,
+                    "consuming_dag_count": 0,
+                },
+                {
+                    "id": 2,
+                    "uri": "s3://bucket/key/2",
+                    "last_dataset_update": None,
+                    "total_updates": 0,
+                    "producing_task_count": 0,
+                    "consuming_dag_count": 0,
+                },
+            ],
+            "total_entries": 2,
+        }
+
+    def test_order_by_raises_400_for_invalid_attr(self, admin_client, session):
+        datasets = [
+            DatasetModel(
+                uri=f"s3://bucket/key/{i}",
+                extra={"foo": "bar"},
+                created_at=timezone.parse(self.default_time),
+                updated_at=timezone.parse(self.default_time),
```

Comment:
```suggestion
```

Not sure we need these?

Reply:
Disagree. The [select and join](https://github.com/apache/airflow/pull/26358/files#diff-917b70d2661fae8322c538a4f60b9d4ba755556036fe0c527eb0411a6ac60f2eR3628-R3650) are not simple, and they rely on aggregating timestamps on related objects. I explicitly included these timestamps here to help keep any errors made in the query loud, instead of potentially silent.

But I'm happy to remove them if you insist.

Reply:
I mean, this test specifically should never even get that far? Plus in general these timestamps on DatasetModel aren't actually used in the query at all?

Reply:
> I mean, this test specifically should never even get that far?

A fair point. :+1:

> Plus in general these timestamps on DatasetModel aren't actually used in the query at all?

While this is true, I think well written fixtures are also useful for trying to catch things that future contributors might mess up. And mistyping something to use `DatasetModel.created_at` instead of `DatasetDagRunQueue.created_at`, for example, is one of them (`updated_at` less so, but I figure it can't really hurt anything other than minute readability of the test code, and something else might have an `updated_at` attribute/column added in the future).

But this is also speculative - I'll remove it, here and elsewhere where it isn't needed.